pkg/types/vsphere/platform.go

```diff
@@ -5,6 +5,9 @@ type Platform struct {
 	// VCenter is the domain name or IP address of the vCenter.
 	VCenter string `json:"vCenter"`
 
+	// Port is the transportation port for vcenter server
+	Port string `json:"port,omitempty"`
```

Comment:
Personally I would like that we allow user to give the vCenter location including the port. Having a separate field for port is not ergonomic.

Reply:
@abhinavdahiya vsphere cloud config has a separate field for port, so if they are combined we will need to `split`. Does that change your mind? It changed mine, but it is not a strong preference.

Reply:
Not really, the entire goal for our API is to make it easy for users to create the cluster, and we should take on complexity whenever it helps remove it for users..